### PR TITLE
[incubator/chartmuseum] Add storage provider configuration options

### PR DESCRIPTION
**Pull request by grue**
I've added some rudimentary configuration options to allow chartmuseum container configs to back themselves against S3, GCS, or a persistent volume. 

Let me know what you think.

### CODE REVIEW

**Comment by cory-klein** [comment]
Where would one provide the auth for the S3 bucket?

**Comment by grue** [comment]
@cory-klein several ways currently: the Node's IAM role or environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY being set.

It also could be possible to mount a volume with the credentials file into /root/.aws in the container's filesystem, but I haven't looked at doing that.

**Comment by grue** [comment]
For IAM roles from kube2iam, I think I need to add support for customization of annotations. I'll look at doing that if that's interesting to you.

**Comment by mattfarina** [comment]
/ok-to-test

cc: @jdolitsky

**Comment by jdolitsky** [comment]
@grue @cory-klein I would like to suggest closing this PR, due to another PR which addresses these changes and more (secrets, ingress, etc): #3050 

My apologies for not reviewing this sooner. Please comment/review #3050 on whether it supports your use case.

Please submit any improvements not present in #3050 (e.g. NOTES.txt changes) as a separate PR. Thanks. Sorry again for the slow feedback. In any future PRs, please CC me and other chartmuseum contributors: https://github.com/kubernetes-helm/chartmuseum/graphs/contributors

File: incubator/chartmuseum/templates/_container.tpl [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{{- define "container" -}}
```
Comment (tmaier) [inline incubator/chartmuseum/templates/_container.tpl:1]: please namespace this, as template definitions are global

**Comment by grue** [comment]
@jdolitsky That sounds good to me, I'll take a look at the other PR's changes. Likewise, my apologies for the slow response.

**Pull request closed without merging**